core/analysis/token_attributes.hpp

```diff
@@ -162,15 +162,28 @@ class attribute_provider_change final : public attribute {
   mutable callback_f callback_{&noop};
 };
 
-// Score threshold can be set by document consumers
+// min score set by document consumers
+// max score set by document producers
 struct score_threshold final : public attribute {
   // DO NOT CHANGE NAME
   static constexpr std::string_view type_name() noexcept {
     return "iresearch::score_threshold";
   }
 
-  score_t value;
-  std::span<const score_t> skip_scores;
+  score_t min;
+  // For disjunction/conjunction it's just sum of sub-iterators max score
+  // For iterator without score it depends on count of documents in iterator
+  // For wanderator it's max score for whole skip-list
+  // TODO(gnusi) write it to the begin of skip-list
+  // TODO(MBkkt) tail_max better here and not affect correctness
+  //  but to support it we need to know max value in the tail blocks.
+  //  Open question: how do it without read next blocks?
+  // score_t list_max;

```

Comment:
without it impossible to choose any pivot (except == len) and set some min threshold to sub iterators